core/lib/server/validateFields.ts

```diff
@@ -4,13 +4,28 @@ import { getJsonSchemaByCoreSchemaType } from "schemas";
 
 import { CoreSchemaType } from "db/public";
 
+import { htmlToProsemirrorServer, prosemirrorToHTMLServer } from "../editor/serialize-server";
+
 const validateAgainstContextEditorSchema = (value: unknown) => {
 	try {
+		if (typeof value === "string") {
+			const node = htmlToProsemirrorServer(value);
+
+			node.check();
+			// return renderNodeToHTML(node);
+			return { success: true, value };
+		}
+
 		const node = baseSchema.nodeFromJSON(value);
+
+		// TODO: reenable this
 		node.check();
-		return true;
-	} catch {
-		return false;
+
+		const html = prosemirrorToHTMLServer(node);
+
+		return { success: true, value: html };
+	} catch (e) {
+		return { success: false, error: e };
 	}
 };
```

Comment:
this is somewhat of a big change, as it turns the `validation` into `parsing + coercion`. i don't think this is incorrect per se, but it's quite a large change.

this is not necessary per se anymore: we could, on form submission, make the client convert the prosemirror tree to html instead. let me know your opinion here!